Kernel/Thread.cpp

```diff
@@ -114,6 +114,17 @@ Thread::Thread(NonnullRefPtr<Process> process)
 
 Thread::~Thread()
 {
+    {
+        // We need to explicitly remove ourselves from the thread list
+        // here. We may get pre-empted in the middle of destructing this
+        // thread, which causes problems if the thread list is iterated.
+        // Specifically, if this is the last thread of a process, checking
+        // block conditions would access m_process, which would be in
+        // the middle of being destroyed.
```

Comment:
This feels like the kind of comment that eventually gets out-of-sync with the code.
But maybe I'm wrong. Anyways, it's nice to have the information right now :) Thanks for fixing this!

Reply:
Yeah I want to get away from checking block conditions every time we call into the scheduler. Which is causing me plenty grief with multiple processors. So yes, eventually I think this will go away.